src/app/utils.ts

```diff
@@ -0,0 +1,108 @@
+import path from 'path'
+import fs from 'fs'
+import os from 'os'
+
+export const UPLOAD_DIR = path.resolve(os.tmpdir(), 'public/uploads')
+export const metadataPath = path.resolve(UPLOAD_DIR, `metadata.json`)
+
+export const createUploadDirIfNotExists = async () => {
+    try {
+        if (!fs.existsSync(UPLOAD_DIR)) {
+            fs.mkdirSync(UPLOAD_DIR)
+        }
+    } catch (err) {
+        console.error('Error creating directory:', err)
+    }
+}
+
+export const loadMetadata = async () => {
+    let metadata: any = []
+    if (fs.existsSync(metadataPath)) {
+        metadata = JSON.parse(fs.readFileSync(metadataPath, 'utf8'))
+    }
+    return metadata
+}
+
+export const getFileExtension = (filename: string) => {
+    const parts = filename.split('.')
+    return parts.length > 1 ? parts.pop()?.toUpperCase() : 'N/A'
+}
+
+// TODO Override runID if already exists
+export const saveMetadata = async (file: Blob, name: string, runId: string) => {
```

Comment:
I don't know if all of this is necessary. Just trying to reduce the amount of code we need to maintain and keep in sync (e.g. the file content might not actually match up with the file size in the metadata or the timestamp on the file might not match the time stored in this metadata file)

Reply:
We can remove the file size. The timestamp here is for when the file was uploaded and not the file timestamp and is currently displayed in the table that lists the results. 
![image](https://github.com/user-attachments/assets/b32109f6-ed2d-4214-88c6-7248c952bec8)